src/main/resources/templates/registration.html

```diff
@@ -110,6 +117,10 @@ <h1 th:text="#{label.form.title}">form</h1>
         {
             window.location.href = serverContext + "emailError.html";
         }
+        else if(data.responseJSON.error == "InvalidReCaptcha"){
+            $("#captchaError").show().html(data.responseJSON.message);
+            grecaptcha.reset()
+        }
         else if(data.responseJSON.error == "UserAlreadyExist"){
             $("#emailError").show().html(data.responseJSON.message);
```

Comment:
I moved the reset to the top of the error handler.  Is this appropriate?  Since it is the first thing checked, any error would invalidate the token.  When the keys aren't present; the captcha isn't enabled.  Should expression become `grecaptcha && grecaptcha.reset()` ?

I originally thought that you had to check for duplicate tokens. But looking back; I had forgotten to actually check the payload of the request.  I was only validating the status text; which was always 'success'.  When actually the same token wasn't validating twice.

I've since changed the cache to instead count failed response tokens; and blocking them like the LoginAttemptService.  -- I added some detail in the recent article iteration;